src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -807,7 +815,17 @@ private async ValueTask SendRequestContentAsync(HttpRequestMessage request, Http
             if (sendRequestContent)
             {
                 if (NetEventSource.IsEnabled) Trace($"Sending request content for Expect: 100-continue.");
-                await SendRequestContentAsync(request, stream, async, cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    await SendRequestContentAsync(request, stream, async, cancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception) when (timerExpired)
```

Comment:
We're adding a decent amount of complication to be able to pass around both sendRequestContent and timerExpired.  Could we instead just always call Dispose in this case if SendRequestContentAsync throws?  As you said, we're talking about the combination of a faulty request content + Expect: 100-continue.  If we get an exception in that case, regardless of the timer, seems like it'd be fine to just get rid of the connection in that case.